cmd/snap-confine/mount-support.c

```diff
@@ -301,7 +305,13 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 		}
 		sc_must_snprintf(dst, sizeof dst, "%s/%s", scratch_dir,
 				 mnt->path);
-		sc_do_mount(mnt->path, dst, NULL, MS_REC | MS_BIND, NULL);
+		if (mnt->is_optional) {
+			sc_do_optional_mount(mnt->path, dst, NULL,
```

Comment:
The original logic had a "continue" in case of failures. Isn't that required here still?

Reply:
Yeah, I restored that just now.